src/crimson/osd/osd_meta.h

```diff
@@ -31,6 +32,9 @@ class OSDMeta {
     : store{store}, coll{coll}
   {}
 
+  seastar::future<> clear() {
```

Comment:
nit, i'd suggest move this change into a separated change. and explain why we want to clear the OSDMeta in the commit message.